src/components/CoinsSelectAutocomplete/CoinsSelectAutocomplete.tsx

```diff
@@ -1,411 +1,22 @@
-      Icon: 'dot',
-      value: 'dot',
-    },
-    {
-      Title: 'SOL',
-      Description: 'Solana',
-      Icon: 'sol',
-      value: 'sol',
-    },
-    {
-      Title: 'POLYGON',
-      Description: 'POLYGON Chain',
-      Icon: 'polygon',
-      value: 'polygon',
-    },
-    {
-      Title: 'BCHA',
-      Description: 'Bitcoin ABC',
-      Icon: 'xec',
-      value: 'bcha',
-    },
-    {
-      Title: 'DOGE',
-      Description: 'Dogecoin',
-      Icon: 'doge',
-      value: 'doge',
-    },
-    {
-      Title: 'ADA',
-      Description: 'Cardano',
-      Icon: 'ada',
-      value: 'ada',
-    },
-  ] as const,
-  test: [
-    {
-      Title: 'TBTC',
-      Description: 'Testnet Bitcoin',
-      Icon: 'btc',
-      value: 'tbtc',
-    },
-    {
-      Title: 'TXRP',
-      Description: 'Testnet Ripple',
-      Icon: 'xrp',
-      value: 'txrp',
-    },
-    {
-      Title: 'TXLM',
-      Description: 'Testnet Stellar',
-      Icon: 'xlm',
-      value: 'txlm',
-    },
-    {
-      Title: 'GTETH',
-      Description: 'Goerli Testnet Ethereum',
-      Icon: 'eth',
-      value: 'gteth',
-    },
-    {
-      Title: 'GTERC',
-      Description: 'Goerli Testnet ERC20 Token',
-      Icon: 'eth',
-      value: 'gterc20',
-    },
-    {
-      Title: 'TTRX',
-      Description: 'Testnet Tron',
-      Icon: 'trx',
-      value: 'ttrx',
-    },
-    {
-      Title: 'TEOS',
-      Description: 'Testnet Eos',
-      Icon: 'eos',
-      value: 'teos',
-    },
-    {
-      Title: 'TAVAXC',
-      Description: 'Testnet Avalanche C-Chain',
-      Icon: 'avax',
-      value: 'tavaxc',
-    },
-    {
-      Title: 'TNEAR',
-      Description: 'Testnet Near',
-      Icon: 'near',
-      value: 'tnear',
-    },
-    {
-      Title: 'TDOT',
-      Description: 'Testnet Polkadot',
-      Icon: 'dot',
-      value: 'tdot',
-    },
-    {
-      Title: 'TSOL',
-      Description: 'Testnet Solana',
-      Icon: 'sol',
-      value: 'tsol',
-    },
-    {
-      Title: 'TPOLYGON',
-      Description: 'Polygon Mumbai Testnet',
-      Icon: 'polygon',
-      value: 'tpolygon',
-    },
-    {
-      Title: 'TDOGE',
-      Description: 'Dogecoin Testnet',
-      Icon: 'doge',
-      value: 'tdoge',
-    },
-    {
-      Title: 'TADA',
-      Description: 'Cardano Testnet',
-      Icon: 'ada',
-      value: 'tada',
-    },
-  ] as const,
-};
-
 export type CoinsSelectAutocompleteProps = {
   onChange: React.ChangeEventHandler<HTMLSelectElement>;
+  coins: readonly CoinMetadata[];
+  selectedCoin?: string;
+  helperText?: React.ReactNode;
 };
 
 export function CoinsSelectAutocomplete({
   onChange,
+  coins,
+  selectedCoin,
+  helperText,
 }: CoinsSelectAutocompleteProps) {
-  const { env = 'test', coin = '' } = useParams<'env' | 'coin'>();
-  const safeBitgoEnv = safeEnv(env);
-  const isUnsignedSweep = window.location.href.includes('build-unsigned-sweep');
-  const coins = isUnsignedSweep
-    ? buildUnsignedSweepCoins
-    : nonBitgoRecoveryCoins;
```

Comment:
For Wrong Chain Recoveries there will be two coin selects, the second being dependent on the first. As such it will not be maintainable to derive the coins list from this component, it should instead be passed in to the component.